src/stream.iced

```diff
@@ -0,0 +1,26 @@
+stream = require('stream')
+enc = require('./encoding')
+
+desired_block_size = 4096
+calculate_block_size = (input_length) ->
+  input_length*Math.floor(desired_block_size/input_length)
+
+exports.StreamEncoder = class StreamEncoder extends stream.Transform
+
+  constructor : (@encoder) ->
+    super({highWaterMark : calculate_block_size(encoder.in_block_len)})
+
+  _transform : (chunk, encoding, cb) ->
+    @push(@encoder.encode(chunk))
+    @read(0)
```

Comment:
I'm assuming you're talking about the `@read(0)` line - it refreshes the internal read buffer. This is what fixed the truncation bug where we weren't getting everything back after encode->decode operations that went past the highWaterMark. From the node API: https://nodejs.org/api/stream.html#stream_readable_read_0